core/src/forwarding_stage.rs

```diff
@@ -465,6 +507,86 @@ impl<VoteClient: ForwardingClient, NonVoteClient: ForwardingClient>
     }
 }
 
+#[async_trait]
+impl LeaderUpdater for ForwardAddressGetter {
+    fn next_leaders(&mut self, lookahead_slots: usize) -> Vec<SocketAddr> {
+        self.get_non_vote_forwarding_addresses(lookahead_slots as u64, Protocol::QUIC)
+    }
+
+    async fn stop(&mut self) {}
+}
+
+#[derive(Clone)]
+struct TpuClientNextClient {
+    runtime_handle: RuntimeHandle,
+    sender: mpsc::Sender<TransactionBatch>,
+}
+
+const METRICS_REPORTING_INTERVAL: Duration = Duration::from_secs(3);
+
+impl TpuClientNextClient {
+    fn new(
+        runtime_handle: tokio::runtime::Handle,
+        forward_address_getter: ForwardAddressGetter,
+        stake_identity: Option<&Keypair>,
+    ) -> Self {
+        // TODO(klykov): find the best channel size
```

Comment:
This I need to investigate to find the optimal channel size. Not quite sure yet how exactly.

Reply:
Set to 128 for now, add metric to watch true size (you need 2 counters: one that counts inserts, one that adds up channel length, when you report just divide and report mean length). Maybe worth making a wrapper around tokio MPSC that does this automagically (we have EvictingSender) as example of such wrapper.

Reply:
Not that easy to add this metric because ForwardingStage supports both ConnectionCache which doesn't have sender and TPuClientNext. One way would be to add the size of the channels as a metric to the tpu-client-next itself here: https://github.com/anza-xyz/agave/blob/master/tpu-client-next/src/metrics.rs